### PR TITLE
feat(es/minifier): Inline lazy inited vars used once

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**
Mainly for inline var inited with functions, basically a followup for #5404 

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc/tests/tsc-references/stringNamedPropertyAccess.2.minified.js [hunk 1]
```diff
@@ -1,7 +1,4 @@
 //// [stringNamedPropertyAccess.ts]
+var i, a;
 import _class_call_check from "@swc/helpers/src/_class_call_check.mjs";
-var i, a, C = function C() {
```
Comment (kdy1) [inline crates/swc/tests/tsc-references/stringNamedPropertyAccess.2.minified.js:3]: I'm not sure if this is correct.
This file calls `_class_call_check` but the new version does not call it

Comment (kdy1) [reply crates/swc/tests/tsc-references/stringNamedPropertyAccess.2.minified.js:3]: Oh is there `PURE`?

Comment (Austaras) [reply crates/swc/tests/tsc-references/stringNamedPropertyAccess.2.minified.js:3]: No, it got inlined into
```js
(function C() {
    "use strict";
    _class_call_check(this, C);
})["c d"]
```
as access to function property is pure, this got elimintated.

Comment (kdy1) [reply crates/swc/tests/tsc-references/stringNamedPropertyAccess.2.minified.js:3]: I see, thank you!

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc/tests/tsc-references/ClassAndModuleThatMergeWithStaticFunctionAndExportedFunctionThatShareAName.2.minified.js [hunk 2]
```diff
@@ -12,12 +12,9 @@ var A, Point = function() {
         };
     }, Point;
 }();
-!function(Point) {
-    var Origin = function() {
-        return null;
-    };
-    Point.Origin = Origin;
-}(Point || (Point = {})), function(A) {
+(Point || (Point = {})).Origin = function() {
```
Comment (jridgewell) [inline crates/swc/tests/tsc-references/ClassAndModuleThatMergeWithStaticFunctionAndExportedFunctionThatShareAName.2.minified.js:15]: Not sure if we want to maintain the function's `name`

**Review by jridgewell — DISMISSED** [review]

File: crates/swc/tests/tsc-references/ClassAndModuleThatMergeWithStaticFunctionAndExportedFunctionThatShareAName.2.minified.js (hunk 2 above)
Comment (kdy1) [reply crates/swc/tests/tsc-references/ClassAndModuleThatMergeWithStaticFunctionAndExportedFunctionThatShareAName.2.minified.js:15]: It depends on the configuration

**Review by kdy1 — APPROVED** [review]

**Pull request merged**